include/IrcCore/ircdebug_p.h

```diff
@@ -142,8 +145,16 @@ static bool irc_debug_enabled(IrcConnection* c, uint l)
 
         dbg_init = true;
     }
-
-    return l <= dbg_level && (dbg_name.isEmpty() || QRegularExpression(QRegularExpression::wildcardToRegularExpression(dbg_name), QRegularExpression::CaseInsensitiveOption).match(c->displayName()).hasMatch());
+    if (l <= dbg_level && (dbg_name.isEmpty()))
+        return true;
```

Comment:
if (A && B)
        return true;
    return C;

is not the same as

    return A && (B || C);